### PR TITLE
using sp_get and replacing strlen with sizeof

### PR DESCRIPTION
**Pull request by deepak1556**
everything works well in the cli tool and the clear-similar-keys test passes. however assertion of `simple-get-set fails`, can you guys check it out.

### CODE REVIEW

**Comment by deepak1556** [comment]
replaced strcmp with strncmp in `test\simple-get-set.c`

**Comment by deepak1556** [comment]
@stephenmathieson https://github.com/sphia/sphia/commit/5e3d655640756c32928c51d69888b23ab5dd6671

**Comment by jwerle** [comment]
Nice work! @larzconwell @stephenmathieson have you guys tried this yet?

**Comment by stephenmathieson** [comment]
LGTM!

**Comment by jwerle** [comment]
Sweet ! I'll merge when @larzconwell gives the OK too !!

**Comment by stephenmathieson** [comment]
@jwerle yes please!

**Comment by jwerle** [comment]
@stephenmathieson Sweet I'll start putting something together soon. Hmm.. So does this reveal bad use of sp_set/sp_get on our end ? Is this still a bug in sophia?

File: src/get.c [hunk 1]
```diff
@@ -11,22 +11,13 @@
...
-    size_t ksize = strlen(k);
-
-    if (size == ksize) {
-      if (0 == strncmp(key, k, ksize) && 0 == strncmp(k, key, size)) {
-        sp_destroy(_c);
-        return v;
-      }
-    }
-  }
+  rc = sp_get(sphia->db, key, size, &value, &vsize);
```
Comment (stephenmathieson) [inline src/get.c:19]: Is it possible that `value` is not terminated here?

**Comment by stephenmathieson** [comment]
@jwerle I still think this is a sophia bug.  It seems we're just compensating for it by using `strlen(x) + 1`.

File: src/get.c (hunk 1 above)
Comment (stephenmathieson) [inline src/get.c:19]: perhaps we should:

``` c
rc = sp_get(sphia->db, key, size, &value, &vsize);

if (1 == rc) {
  if ('\0' != value[vsize]) value[vsize] = '\0';
  return value;
}

return NULL;
```

Comment (jwerle) [inline src/get.c:19]: That could definitely work, as is much simpler than iterating through every key

**Comment by larzconwell** [comment]
You guys don't have to wait for me, I've been busy so can't do much.

This looks good to me +1

**Comment by jwerle** [comment]
@larzconwell sweet!
@stephenmathieson we good to merge? cc @deepak1556

**Comment by deepak1556** [comment]
:+1: ,Thanks guys for the quick review :)

**Comment by jwerle** [comment]
@deepak1556 np !

**Comment by stephenmathieson** [comment]
Yep, lgtm

**Pull request merged**